components/quasar_s3_export_role/main.tf

```diff
@@ -0,0 +1,67 @@
+resource "aws_s3_bucket" "dosomething_quasar_archive" {
+  bucket = "dosomething-quasar-archive"
+  acl    = "private"
+
+  lifecycle_rule {
+    id      = "archive"
+    enabled = true
+
+    prefix = "archive/"
+
+    tags = {
+      "rule" = "archive"
+    }
+
+    transition {
+      days          = 30
+      storage_class = "STANDARD_IA" # or "ONEZONE_IA"
+    }
+
+    transition {
+      days          = 30
+      storage_class = "GLACIER"
+    }
+
+    expiration {
+      days = 60
+    }
```

Comment:
Does this mean that these archived objects will be deleted after 60 days?

Reply:
Yes, they'll expire after 60 days. Would you prefer a longer time period?

Reply:
I think I'd misunderstood this task, since I'd assumed the reason we were putting these into Glacier was for long-term storage. Is this something you're planning on using for RDS's daily snapshots?

Reply:
I was thinking that we'd push the snapshots into s3 daily and then push them into Glacier after a set period, and eventually expire them since we already have the backup in Glacier if we really need to pull it out.

Reply:
I believe the Glacier lifecycle rule "moves" those objects into Glacier (as their storage class in S3), but the objects themselves are still represented as S3 items. So in this case, these rules would be moving the objects into Glacier and then deleting them (from both S3 & Glacier) 30 days later.

Reply:
If we want to store these indefinitely in Glacier, we don't need an `expiration` rule here. If we _do_ want to expire these so these backups don't pile up over time, we should set an `expiration` rule of [at least 90 days](https://docs.aws.amazon.com/AmazonS3/latest/dev/lifecycle-transition-general-considerations.html#before-deciding-to-archive-objects) or else it looks like we'll be charged an "early deletion fee".